src/promptflow/tests/sdk_cli_azure_test/conftest.py

```diff
@@ -172,11 +178,14 @@ def single_worker_thread_pool() -> None:
     def single_worker_thread_pool_executor(*args, **kwargs):
         return ThreadPoolExecutor(max_workers=1)
 
-    with patch(
-        "promptflow.azure.operations._run_operations.ThreadPoolExecutor",
-        new=single_worker_thread_pool_executor,
-    ):
+    if is_live():
```

Comment:
It's weird to see so many `is_live()` judgement here and there, couldn't we put it at some entry place to avoid such duplicate usages?

Reply:
actually, I think we cannot avoid this, because we don't expect a mono fixture that contains every mock in it, so we need many fixtures; and we expect the mock only happen during non-live mode - so we get it everywhere.

Reply:
and I have just taken a look at dpv2 repository: [conftest.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/tests/conftest.py), also everywhere:

![image](https://github.com/microsoft/promptflow/assets/38847871/fdbbcc31-83f7-46c9-bd3e-b5c4124863bd)